date_range/models/date_range.py

```diff
@@ -28,11 +28,45 @@ def _default_company(self):
     active = fields.Boolean(
         help="The active field allows you to hide the date range without "
         "removing it.", default=True)
+    parent_type_id = fields.Many2one(
+        related='type_id.parent_type_id',
+        store=True,
+        readonly=True)
+    parent_id = fields.Many2one(
+        comodel_name='date.range', string="Parent",
+        domain="['|', ('type_id.parent_type_id', '!=', parent_type_id), "
+        "('parent_type_id', '=', False)]",
+        index=1)
 
     _sql_constraints = [
         ('date_range_uniq', 'unique (name,type_id, company_id)',
          'A date range must be unique per company !')]
 
+    @api.constrains('parent_id', 'date_start', 'date_end')
+    def _validate_child_range(self):
+        for this in self:
+            if not this.parent_id:
+                continue
+            start = this.parent_id.date_start <= this.date_start
+            end = this.parent_id.date_end >= this.date_end
+            child_range = start and end
+            if not child_range:
+                if (not start) and end:
```

Comment:
These messages will be seen by normal users. The word 'compatible' is really tech lingo. Better say something like:
```
_("Start date %(start) of %(name)s must be greater than or equal to start date %(parent_start)s of %(parent_name)s") %
{'start': this.date_start, 'name': this.name,
'parent_name': this.parent_id.name, 'parent_start': this.parent_id.date_start}
```